⚡️ Speed-up race-condition schedulers (#3891)
dubzzz/fast-check: website/docs/tutorials/detect-race-conditions/one-step-closer-to-real-usages.mdx

await s.waitFor(Promise.all(pendingQueries));
```

> An alternative would have been to use `waitAll` instead of `waitFor` but it comes with a precise requirement: promises have to be imperatively scheduled by the time we reach them. In other words, if our code delays a little bit the call to `call`, the scheduler might not wait enough.

Author: ```suggestion
An alternative would have been to use `waitAll` instead of `waitFor` but it comes with a precise requirement: promises have to be already scheduled by the time we request the scheduler. In other words, if our code delays a little bit the call to `call`, the scheduler might not wait enough.
```

> As the fact that `call` has to be fired synchronously when issuing a call on a clean context is not a requirement for our current function, we can relax the constraint in our test to make evolving this implementation easier.

Author: ```suggestion
Given the fact that `call` being fired synchronously is not a requirement for our current function, we can relax the constraint in our test to make evolving this implementation easier.
```
